source/server-discovery-and-monitoring/tests/single/direct_connection_replicaset.yml

```diff
@@ -0,0 +1,23 @@
+description: Direct connection with replicaSet URI option
+
+uri: "mongodb://a/?replicaSet=rs&directConnection=true"
+
+phases:
+  # We are connecting to a replica set member
+  - responses:
+    -
+      - "a:27017"
+      - ok: 1
+        ismaster: true
+        primary: true
```

Comment:
This field should actually contain the hostname of the primary, not a boolean. This caused some issues for the Rust driver's type-aware test file deserializer.

Reply:
I removed this field because it was not used in the "Discover primary with directConnection URI option" test, is this OK?

Reply:
Yep lgtm